web/src/routes/Event.jsx

```diff
@@ -8,19 +13,76 @@ import { Table, Thead, Tbody, Th, Tr, Td } from '../components/Table';
 export default function Event({ eventId }) {
   const apiHost = useApiHost();
   const { data, status } = useEvent(eventId);
+  const [showDialog, setShowDialog] = useState(false);
+  const [deleteStatus, setDeleteStatus] = useState(FetchStatus.NONE);
+  const [deleteMessage, setDeleteMessage] = useState();
+
+  const handleClickDelete = () => {
+    setShowDialog(true);
+  };
+
+  const handleDismissDeleteDialog = () => {
+    setShowDialog(false);
+  };
+
+
+  const handleClickDeleteDialog = useCallback(async () => {
+
+    setDeleteStatus(FetchStatus.LOADING);
+    let success;
+    try {
+      const response = await fetch(`${apiHost}/api/events/${eventId}`, { method: 'DELETE' });
+      const deleteEvent = await getJSON(response)
+      success = deleteEvent.success;
+      setDeleteStatus(success ? FetchStatus.LOADED : FetchStatus.ERROR);
+      setDeleteMessage(deleteEvent.message);
+    } catch (e) {
+      setDeleteStatus(FetchStatus.ERROR);
+    }
+
+    async function getJSON(response) {
+      if (response.status === 204) return {success: true};
+      return response.json();
+    }
```

Comment:
The ordering here is a little odd. Since this function isn't reused, consider removing and inlining the logic agove

Reply:
"response.json();" the .json is a promise, so you have to resolve the promise. Agree ordering is kinda whacky, JS is a third language for me, so still working through some of the weirdness of the lang.  Do you have suggestions?

Reply:
It's simpler than you think :)

```
const { success } = await (response.status < 300 ? response.json() : { success: true });